File: LeetCode/167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input Array Is Sorted
#

# @lc code=start
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        val=int(target/2)
        left,right,mid=0,len(numbers),0
        length=right
        while left<=right:
            print(left," ",right," ",mid)
            mid=int((left+right)/2)
            if numbers[mid]<val:
                left=mid+1
            else: right=mid-1
        right=left
        while left>=0 and right<length:
            sum=numbers[left]+numbers[right]
            if sum==target and left!=right:
                return [left+1,right+1]
            elif sum>target: 
                left-=1
            else: right+=1
            print(left," ",right)
# @lc code=end

File: LeetCode/167.two-sum-ii-input-array-is-sorted.cpp
/*
 * @lc app=leetcode id=167 lang=cpp
 *
 * [167] Two Sum II - Input Array Is Sorted
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int len=nums.size(),i,j;
        vector<int> a(2,0);
        for (i=0;i<len;i++){
            for (j=i+1;j<len;j++){
                int sum=nums[i]+nums[j];
                if (sum==target){
                    a[0]=i+1;
                    a[1]=j+1;
                    return a;
                }else if (sum>target){
                    continue;
                }
            }
        }
        return a;
    }
};
// @lc code=end

